Bottom: 208f7ed73cbd38b0a5da0a22cf473eddfbd04558
Top:    14e00c097f48c6d8fd95980649795b008ad6e6ae
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-08-14 15:30:30 -0500

Configure freenode as IRC reporter


---
diff --git a/docker/buildbot-master/Dockerfile b/docker/buildbot-master/Dockerfile
index a79778ef..d9b8f54b 100644
--- a/docker/buildbot-master/Dockerfile
+++ b/docker/buildbot-master/Dockerfile
@@ -19,7 +19,8 @@ RUN pip --no-cache-dir install \
     'buildbot-grid-view' \
     'buildbot-waterfall-view' \
     'buildbot-www' \
-    'ldap3'
+    'ldap3' \
+    'pyOpenSSL'
 
 # Create buildbot user
 ARG BUILDBOT_UID=1000
diff --git a/docker/buildbot-master/buildbot/master.cfg b/docker/buildbot-master/buildbot/master.cfg
index 5f1c821e..1ac46a1a 100644
--- a/docker/buildbot-master/buildbot/master.cfg
+++ b/docker/buildbot-master/buildbot/master.cfg
@@ -5,7 +5,8 @@
 # 'master.cfg' in your buildmaster's base directory.
 
 import os
-from buildbot.plugins import secrets,steps,util
+import json
+from buildbot.plugins import reporters, secrets, steps, util
 from urlparse import urlparse
 
 os.environ['LONG_RUN_TIMEOUT']='14400'
@@ -34,10 +35,17 @@ c['secretsProviders'] = [secrets.SecretInAFile(dirname="/run/secrets")]
 
 c['services'] = []
 
-from buildbot.plugins import reporters
-irc = reporters.IRC(host="172.16.64.3",
-                    nick="orionbb",
-                    channels=["#orion"])
+irc = reporters.IRC(
+    allowForce=False,
+    allowShutdown=False,
+    channels=json.loads(os.getenv("BUILDBOT_IRC_CHANNELS")),
+    host=os.getenv("BUILDBOT_IRC_HOST"),
+    nick=os.getenv("BUILDBOT_IRC_NICK"),
+    notify_events=[],
+    password=os.getenv("BUILDBOT_IRC_PASS"),
+    useColors=True,
+    useSSL=True
+)
 c['services'].append(irc)
 
 mn = reporters.MailNotifier(
